pkg/types/types.go

```diff
@@ -270,24 +284,84 @@ func (s byStepTime) Less(i, j int) bool {
 	return s[i].StepTime < s[j].StepTime
 }
 
-func mergeMetrics(metrics []Metric, mismatchCheck bool) (metric Metric, mismatches int) {
+func areFloatsEqual(a, b float64) bool {
+	epsilon := math.Nextafter(1.0, 2.0) - 1.0
+	floatMinNormal := math.Float64frombits(0x0010000000000000)
+
+	absA := math.Abs(a)
+	absB := math.Abs(b)
+	diff := math.Abs(a - b)
+
+	if a == b {
+		return true
+	} else if a == 0 || b == 0 || (absA+absB < floatMinNormal) {
+		return diff < (epsilon * floatMinNormal)
+	} else {
+		div := math.MaxFloat64
+		if div > absA+absB {
+			div = absA + absB
+		}
+		return diff/div < epsilon
+	}
+}
+
+func getPointMajorityValue(values []float64) (majorityValue float64, majorityCount int, err error) {
+	valuesCount := len(values)
+	if valuesCount == 0 {
+		return math.NaN(), 0, fmt.Errorf("no value for majority voting")
```

Comment:
It is very strange to use `NaN` in case of incorrect request. `NaN` maybe not be processed correctly all the time. It is really special. I think using a default type value is preferable.

Reply:
So the thing is that we are returning an error at the same time. so it doesn't matter much what we are returning for the values. I understand your concern, but I think we can go ahead.

Reply:
Anyways, decided to make the change :D